flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotProvider.java

```diff
@@ -91,6 +92,20 @@
 			allocationTimeout);
 	}
 
+	/**
+	 * Allocates a bulk of physical slots. The allocation will be completed
+	 * normally only when all the requests are fulfilled.
+	 *
+	 * @param physicalSlotRequests requests for physical slots
+	 * @param timeout indicating how long it is accepted that the slot requests can be unfulfillable
+	 * @return future of the results of slot requests
+	 */
+	default CompletableFuture<Collection<PhysicalSlotRequest.Result>> allocatePhysicalSlots(
```

Comment:
I think that might not be good since we will still need to declare the default body here to avoid implement this method for all `SlotProvider` implementations .

Reply:
I think having default method is fine. It is more related to making `OneSlotPerExecutionSlotAllocator` depending only on `BulkSlotProvider `.

Reply:
Ok. It would be good to limit `OneSlotPerExecutionSlotAllocator` to only use methods of the new interface.

Reply:
done.